db/document.go

```diff
@@ -515,6 +515,9 @@ func parseXattrStreamData(xattrName string, userXattrName string, data []byte) (
 	}
 
 	xattrsLen := binary.BigEndian.Uint32(data[0:4])
+	if int(xattrsLen) > len(data) {
+		return nil, nil, nil, fmt.Errorf("%w (%d) from bytes %s", base.ErrXattrInvalidLen, xattrsLen, data)
+	}
```

Comment:
Are we able to write a test for what happens if data[0:4] doesn't represent xattrsLen (i.e. document doesn't have xattrs, it's just the body), but the Uint32 value of data[0:4] is less than len(data)?  I'm assuming we'll return an error while trying to parse the xattr key/value pairs below, but would be good to verify there aren't any subsequent panics in that case.

Reply:
Added a test for this.